src/olympia/amo/middleware.py

```diff
@@ -88,6 +90,20 @@ def process_request(self, request):
         request.LANG = prefixer.locale
 
 
+class AuthenticationMiddlewareWithoutAPI(AuthenticationMiddleware):
+    """
+    Like AuthenticationMiddleware, but disabled for the API, which uses its
+    own authentication mechanism.
+    """
+    def process_request(self, request):
+        if request.path.startswith('/api/'):
+            request.user = AnonymousUser()
+        else:
+            return super(
+                AuthenticationMiddlewareWithoutAPI,
+                self).process_request(request)
```

Comment:
I really find the indentation and line-breaking super-weird but if the `super(..., self)` part doesn't fit on the line, hrm...